src/components/App/OutputSection.tsx

```diff
@@ -1,97 +1,169 @@
 import Link from 'next/link';
-import React from 'react';
-import { AiOutlineArrowDown } from 'react-icons/ai';
-import { IoIosCopy, IoIosShareAlt } from 'react-icons/io';
-import { LuQrCode } from 'react-icons/lu';
+import QRCode from 'qrcode.react';
+import React, { useState } from 'react';
+import { FaCheck, FaRegCopy } from 'react-icons/fa';
+import { FaWhatsapp } from 'react-icons/fa6';
+import { FaDiscord, FaLinkedin, FaXTwitter } from 'react-icons/fa6';
+import { HiOutlineDownload } from 'react-icons/hi';
+import { PiShareFatBold } from 'react-icons/pi';
 
 import Button from '@/components/Button';
-import QRCodeModal from '@/components/QRCodeModal';
-import { removeProtocol } from '@/constants/constants';
+import {
+    discordShareUrl,
+    linkedinShareUrl,
+    removeProtocol,
+    twitterShareUrl,
+    whatsappShareUrl,
+} from '@/constants/constants';
 
 import OutputSectionShimmer from '../ShimmerEffect/OutputSectionShimmer';
 
+const RDSIcon = '_next/image?url=%2Frds.png&w=64&q=75';
+
 interface OutputSectionProps {
     originalUrl: string;
     shortUrl: string;
     isLoaded: boolean;
     handleCreateNew: () => void;
-    handleCopyUrl: () => void;
 }
 
-const OutputSection: React.FC<OutputSectionProps> = ({
-    shortUrl,
-    originalUrl,
-    isLoaded,
-    handleCopyUrl,
-    handleCreateNew,
-}) => {
-    const [showQRCodeModal, setShowQRCodeModal] = React.useState<boolean>(false);
+const OutputSection: React.FC<OutputSectionProps> = ({ shortUrl, isLoaded }) => {
+    const [downloaded, setDownloaded] = useState(false);
+    const [copied, setCopied] = useState(false);
+
+    const CopyActionIcon = copied ? FaCheck : FaRegCopy;
+    const DownloadActionIcon = downloaded ? FaCheck : HiOutlineDownload;
+    const DownloadButtonText = downloaded ? 'Downloaded' : 'Download';
+
     if (!isLoaded) {
-        return <OutputSectionShimmer />;
+        return <OutputSectionShimmer data-testid="output-section-shimmer" />;
     }
 
+    const handleDownload = () => {
+        const canvas = document.getElementById('qr-code') as HTMLCanvasElement;
+        if (canvas) {
+            const pngUrl = canvas.toDataURL('image/png').replace('image/png', 'image/octet-stream');
+            const downloadLink = document.createElement('a');
+            downloadLink.href = pngUrl;
+            downloadLink.download = `${shortUrl.split('/').pop()}.png`;
+            document.body.appendChild(downloadLink);
+            downloadLink.click();
+            document.body.removeChild(downloadLink);
+            setDownloaded(true);
+        }
```

Comment:
- instead of nesting this can we do an early return here ? 
```ts
if (!canvas) return

.... rest of the logic
```

Reply:
Fixed